src/spatialexperiment/_validators.py

```diff
@@ -0,0 +1,86 @@
+import biocframe
+import biocutils as ut
+
+__author__ = "keviny2"
+__copyright__ = "keviny2"
+__license__ = "MIT"
+
+
+def _validate_spatial_coords_names(spatial_coords_names, spatial_coords):
+    if not ut.is_list_of_type(spatial_coords_names, str):
+        raise TypeError("'spatial_coords_names' is not a list of strings")
+
+    if len(spatial_coords_names) != spatial_coords.shape[1]:
+        raise ValueError(
+            f"Expected {spatial_coords.shape[1]} names. Got {len(spatial_coords_names)} names."
+        )
+
+
+def _validate_column_data(column_data, img_data):
+    error_message = "'column_data' must have a column named 'sample_id'."
+
+    if column_data is None:
+        raise ValueError(error_message)
+
+    if not isinstance(column_data, biocframe.BiocFrame):
+        raise TypeError("'column_data' must be a BiocFrame object.")
+    
+    if "sample_id" not in column_data.columns:
+        raise ValueError(error_message)
+
+    if column_data.shape[0] == 0 or img_data.shape[0] == 0:
+        return
+
+    num_unique_sample_ids = len(img_data["sample_id"].unique())
+    num_unique_sample_ids_provided = len(column_data["sample_id"].unique())
+
+    if num_unique_sample_ids != num_unique_sample_ids_provided:
+        raise ValueError(f"Number of unique 'sample_id's is {num_unique_sample_ids}, but {num_unique_sample_ids_provided} were provided.")
```

Comment:
In addition, should check if all `sample_id`'s mentioned in the `column_data` actually exist in `img_data`.